test/XRP/issued-currency-client-integration.test.ts

```diff
@@ -94,38 +133,24 @@ describe('IssuedCurrencyClient Integration Tests', function (): void {
     )
 
     // THEN the transaction was successfully submitted and the correct flag was set on the account.
-    const transactionHash = result.hash
-    const transactionStatus = result.status
-
-    // get the account data and check the flag bitmap
-    const networkClient = new GrpcNetworkClient(rippledGrpcUrl)
-    const account = networkClient.AccountAddress()
-    const classicAddress = XrpUtils.decodeXAddress(wallet.getAddress())
-    account.setAddress(classicAddress!.address)
-
-    const request = networkClient.GetAccountInfoRequest()
-    request.setAccount(account)
-
-    const ledger = new LedgerSpecifier()
-    ledger.setShortcut(LedgerSpecifier.Shortcut.SHORTCUT_VALIDATED)
-    request.setLedger(ledger)
-
-    const accountInfo = await networkClient.getAccountInfo(request)
-    if (!accountInfo) {
-      throw XrpError.malformedResponse
-    }
-
-    const accountData = accountInfo.getAccountData()
-    if (!accountData) {
-      throw XrpError.malformedResponse
-    }
-
-    const flags = accountData.getFlags()?.getValue()
-
-    assert.exists(transactionHash)
-    assert.equal(transactionStatus, TransactionStatus.Succeeded)
-    assert.isTrue(
-      AccountRootFlag.checkFlag(AccountRootFlag.LSF_REQUIRE_AUTH, flags!),
+    await runFlagIntegrationTestSuccess(
+      wallet,
+      result,
+      AccountRootFlag.LSF_REQUIRE_AUTH,
+    )
+  })
+
+  it('disallowIncomingXrp - rippled', async function (): Promise<void> {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account
+    // WHEN disallowIncomingXrp is called
+    const result = await issuedCurrencyClient.disallowIncomingXrp(wallet)
+
+    // THEN the transaction was successfully submitted and the correct flag was set on the account.
+    await runFlagIntegrationTestSuccess(
+      wallet,
+      result,
+      AccountRootFlag.LSF_DISALLOW_XRP,
     )
```

Comment:
How do you feel about creating another integration test that attempts to send an XRP transaction to this account that has set `disllowIncomingXrp`?  We'd basically want to just create a second faucet wallet and then try to use the `send` method on the `XrpClient` to send an XRP transaction to this account.  Ideally the transaction result will fail (I'm also just curious to understand what happens in this case and check that the SDK handles it correctly and returns an informative transaction status).  Note that the `send` method only returns a transaction hash, and then you have to follow up with `getPaymentStatus` to see the full `TransactionResult` object (I think).  We want to refactor that (so that `send` returns a full `TransactionResult`) cause it's old trash but requires a deprecation so... we're getting around to it.

Reply:
I tried writing that, but the transaction succeeds for some reason. I don't have a good enough understanding of the system to understand why.

Reply:
Note for posterity: As we just discovered together, the docs here https://xrpl.org/accountset.html#accountset-flags indicate that this flag isn't actually enforced by rippled, but rather is an indicator to client applications.  Therefore we can't implement this type of integration test.